src/snowcli/cli/common/sql_execution.py

```diff
@@ -3,14 +3,28 @@
 import logging
 from contextlib import contextmanager
 from functools import cached_property
-from textwrap import dedent
+from io import StringIO
+from textwrap import dedent, indent
+from typing import Iterable
 
 from click import ClickException
 from snowcli.cli.common.cli_global_context import cli_context
-from snowflake.connector.cursor import DictCursor
+from snowflake.connector.cursor import DictCursor, SnowflakeCursor
 from snowflake.connector.errors import ProgrammingError
 
 
+class _SnowcliCursor(SnowflakeCursor):
+    def execute(self, *args, **kwargs):
+        """This allows us to show failing query to users."""
+        # TODO: Use exception annotation once we support Python 3.11
+        try:
+            super().execute(*args, **kwargs)
+        except Exception:
+            print("Query:")
+            print(indent(args[0], "    "))
```

Comment:
Do you think we should think about security of printing the query always, even if CLI is not in `verbose` / `debug` mode?

Reply:
Interesting point, there's definitely is some security concern. Using verbose/debug already enables query logging.